### DISCUSSION TITLE
Use a subpath without nginx or something similar

### DISCUSSION THREAD

**Discussion by JVT038** [miguelgrinberg/Flask-SocketIO#1766] [General]
Hi,
So I want to run a SocketIO application without the use of NGINX or something similar to it.
I'm just running the instance like this: `socketio.run(app, str(host), int(port), log_output=strtobool(str(log_output)))`, and I would like to run the instance on `127.0.0.1:5000/subpath` instead of only the `address:port`.
Is it possible to do this, without NGINX?

**Comment by miguelgrinberg** [comment]
Socket.IO always runs on a subpath. The default location of the Socket.IO endpoint is on the `/socket.io` path. You never see this because neither the client nor the server require you to enter this path, it is just appended to the server URL that your application uses.

If you want to change this path, you can, but it is not done by changing the connection URL. In the JavaScript and Python implementations of Socket.IO, the path of the Socket.IO endpoint is given in a `path` option, separate from the server specification.

**Reply by JVT038** [reply]
I believe you misunderstood me.
What I meant was that I could actually access the website / Flask application on a subpath.
And since the server / Flask application is started with the `socketio.run()` function, I decided to ask here.
So I want to visit my Flask app on `127.0.0.1:5000/subpath`.

**Reply by miguelgrinberg** [reply]
There are a few ways to do this, but one that I find very easy is to put all the application routes inside a blueprint, then you can register the blueprint with a `url_prefix`.

**Reply by JVT038** [reply]
Thanks, it worked!
